profiler: Explain splunk.profiler.enabled behavior in readme (#713)
signalfx/splunk-otel-java: profiler/README.md

|`splunk.profiler.tracing.stacks.only`     | false                  | set to `true` to include only stack traces that are linked to a span context |
|`splunk.profiler.period.{eventName}`      | n/a                    | DEPRECATED. Use `splunk.profiler.call.stack.interval` instead.

> If `splunk.profiler.enabled` option is not set to `true`, all profiling features will be disabled,
> even if something else is explicitly enabled. For example, setting `splunk.profiler.memory.enabled`
> to `true` will have no effect in that case. There is no separate toggle for periodic collection of
> call stacks (thread dumps) - this is always enabled when profiler is enabled.

Reviewer: ```suggestion
If the `splunk.profiler.enabled` option is not enabled, all profiling features are disabled. For 
example, setting `splunk.profiler.memory.enabled` to `true`has no effect if `splunk.profiler.enabled`
is set to `false`. Similarly, there is no separate toggle for periodic collection of call stacks (thread dumps),
as this feature is enabled only when the profiler is enabled.
```
